test/object-store/realm.cpp

```diff
@@ -1079,6 +1080,21 @@ TEST_CASE("SharedRealm: convert") {
         REQUIRE(sync_realm->read_group().get_table("class_object")->size() == 1);
     }
 
+    SECTION("cannot convert a local flx sync") {
+        using namespace realm::app;
+        auto make_schema = [] {
+            Schema schema{{"Obj",
+                           {{"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+                            {"value", PropertyType::Int | PropertyType::Nullable}}}};
+
+            return FLXSyncTestHarness::ServerSchema{std::move(schema), {"value"}};
+        };
+        FLXSyncTestHarness harness("c_api_flx_sync_test", make_schema());
```

Comment:
Tests which talk to baas need to be guarded with the appropriate define, and this is failing to build because it's missing. This test also doesn't need to talk to baas; it can just set `flx_sync_requested` on the config and run in .1ms instead of 5 seconds.

Reply:
Agreed, much better like this.